test/e2e/v1beta1/scripts/gh-actions/run-e2e-experiment.py

```diff
@@ -0,0 +1,293 @@
+    # For the LongRunning resume policy Suggestion must be always Running.
+    for c in suggestion.status.conditions:
+        if (
+            c.type == constants.EXPERIMENT_CONDITION_SUCCEEDED
+            and c.status == constants.CONDITION_STATUS_TRUE
+            and experiment.spec.resume_policy == "LongRunning"
+        ):
+            raise Exception(
+                f"Suggestion is Succeeded while Resume Policy is {experiment.spec.resume_policy}."
+                f"Suggestion conditions: {suggestion.status.conditions}"
+            )
+        elif (
+            c.type == constants.EXPERIMENT_CONDITION_RUNNING
+            and c.status == constants.CONDITION_STATUS_TRUE
+            and experiment.spec.resume_policy != "LongRunning"
+        ):
+            raise Exception(
+                f"Suggestion is Running while Resume Policy is {experiment.spec.resume_policy}."
+                f"Suggestion conditions: {suggestion.status.conditions}"
+            )
+
+    # For Never and FromVolume resume policies verify Suggestion's resources.
+    if (
+        experiment.spec.resume_policy == "Never"
+        or experiment.spec.resume_policy == "FromVolume"
+    ):
+        resource_name = exp_name + "-" + experiment.spec.algorithm.algorithm_name
+
+        # Suggestion's Service and Deployment should be deleted.
+        config.load_kube_config()
+        for _ in range(10):
+
+            services = client.CoreV1Api().list_namespaced_service(exp_namespace)
+            for i in services.items:
+                is_deleted = 0 if i.metadata.name == resource_name else 1
+
+            deployments = client.AppsV1Api().list_namespaced_deployment(exp_namespace)
+            for i in deployments.items:
+                is_deleted *= 0 if i.metadata.name == resource_name else 1
+            if is_deleted == 1:
+                break
+
+            # Deployment and Service deletion might take some time.
+            time.sleep(1)
+
+        if is_deleted == 0:
+            raise Exception(
+                "Suggestion Service or Deployment is still alive for "
+                f"ResumePolicy == {experiment.spec.resume_policy}. "
+                f"Alive Services: {[i.metadata.name for i in services.items]}. "
+                f"Alive Deployments: {[i.metadata.name for i in deployments.items]}."
+            )
+
+        # For FromVolume resume policy PVC should not be deleted.
+        if experiment.spec.resume_policy == "FromVolume":
+            PVCs = client.CoreV1Api().list_namespaced_persistent_volume_claim(
+                exp_namespace
+            )
+            is_deleted = 1
+            for i in PVCs.items:
+                if i.metadata.name == resource_name:
+                    is_deleted = 0
+            if is_deleted == 1:
+                raise Exception(
+                    "PVC is deleted for FromVolume resume policy. "
+                    f"Alive PVCs: {[i.metadata.name for i in PVCs.items]}."
+                )
```

Comment:
How about change this lines to use `read_namespaced_persistent_volume_claim(resource_name, exp_namespace)` and check whether this method raise the 404 not found exception or not. And then raise our Exception ?

Since I think i have to read this source code more carefully to understand what this source code wants to test.

Reply:
Sure! I am not sure why Kubernetes API Python client named `get` API for Customer Resources and `read` API for Core Kubernetes Resources.